airbyte-integrations/connectors/source-s3/source_s3/v4/legacy_config_transformer.py

```
@@ -105,12 +122,26 @@ def _transform_file_format(cls, format_options: Union[CsvFormat, ParquetFormat,
             return {"filetype": "parquet", "decimal_as_float": True}
         else:
             # This should never happen because it would fail schema validation
-            raise ValueError(f"Format filetype {format_options} is not a supported file type")
+            raise ValueError(
+                f"Format filetype {format_options} is not a supported file type"
+            )
 
     @classmethod
-    def parse_config_options_str(cls, options_field: str, options_value: Optional[str]) -> Mapping[str, Any]:
+    def parse_config_options_str(cls, options_field: str, options_value: Optional[str]) -> Dict[str, Any]:
         options_str = options_value or "{}"
         try:
             return json.loads(options_str)
         except json.JSONDecodeError as error:
             raise ValueError(f"Malformed {options_field} config json: {error}. Please ensure that it is a valid JSON.")
+
+    @staticmethod
+    def _filter_legacy_noops(advanced_options: Dict[str, Any]):
+        ignore_all = ("auto_dict_encode", "timestamp_parsers")
+        ignore_by_value = (("check_utf8", False),)
```

nice. good idea not to silently ignore `check_utf=True`